helper/helper/src/main/java/com/linkedin/avroutil1/compatibility/RecordGenerationContext.java

```diff
@@ -17,11 +17,14 @@
 public class RecordGenerationContext {
   private final RecordGenerationConfig config;
 
+  private final int MAX_DEPTH = 100;
+  private boolean hasHitMaxDepth = false;
```

Comment:
whats the value of memoizing this now, given that the computation is fairly trivial?

Reply:
We may need to make this a bit more complex though...we should only stop generation for recursive schemas, not all schemas

Reply:
added check directly on recursive schema depth (and memoized it)